test/js/buffer/icon_vertex_buffer.test.js

```diff
@@ -1,16 +1,14 @@
 'use strict';
 
 var test = require('prova');
-var Buffer = require('../../../js/data/buffer/buffer');
 var IconVertexBuffer = require('../../../js/data/buffer/icon_vertex_buffer');
 
 test('IconVertexBuffer', function(t) {
-    var buf = new Buffer();
-    t.ok(new IconVertexBuffer(buf), 'default buffer');
-    var iconBuffer = new IconVertexBuffer(buf);
+    t.ok(new IconVertexBuffer(), 'default buffer');
+    var iconBuffer = new IconVertexBuffer();
     iconBuffer.setupViews();
     t.equal(iconBuffer.itemSize, 16);
-    t.equal(iconBuffer.add(0, 0, 0, 0, 0, 0, 0, 0, [0, 0], 0, 0), undefined);
+    t.equal(iconBuffer.add(0, 0, 0, 0, 0, 0, 0, 0, 0), undefined);
```

Comment:
So the add that I have in the buffer2 icon buffer follows the same interface as the old icon interface. Buffer2 has a lot more validation, so I'm pretty sure these tests would have been failing if we were doing the validation we're doing now.